containers.go

```diff
@@ -377,11 +377,18 @@ func (r *containerStore) removeName(container *Container, name string) {
 	container.Names = stringSliceWithoutValue(container.Names, name)
 }
 
-func (r *containerStore) SetNames(id string, names []string) error {
+func (r *containerStore) SetNames(id string, names []string, opts SetNameOptions) error {
 	names = dedupeNames(names)
 	if container, ok := r.lookup(id); ok {
 		for _, name := range container.Names {
-			delete(r.byname, name)
+			if opts.AppendOnly {
+				names = append(names, name)
+			} else {
+				delete(r.byname, name)
+			}
+		}
+		if opts.AppendOnly {
+			names = dedupeNames(names)
```

Comment:
I think you can drop the first dedupe at line 381 and just do it here not guarded by `opts.AppendOnly`.

Reply:
yes. that would be better.